### PR TITLE
Shares search filter

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Adds a text search filter to the 'Shared with me'-page and aligns the other search inputs to the right.

## Related Issue

- Fixes https://github.com/owncloud/web/issues/10014

## Screenshot
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

### CODE REVIEW

File: packages/web-app-files/src/views/shares/SharedWithMe.vue [hunk 1]
```diff
@@ -166,9 +182,25 @@ export default defineComponent({
...
+        const usersSearchEngine = new Fuse(result, { ...defaultFuseOptions, keys: ['name'] })
+        result = usersSearchEngine.search(unref(filterTerm)).map((r) => r.item)
+      }
+
       return result
     })
 
+    watch(filteredItems, () => {
+      if (!unref(areResourcesLoading)) {
+        markInstance.value = new Mark('.oc-resource-details')
```
Comment (JanAckermann) [inline packages/web-app-files/src/views/shares/SharedWithMe.vue:195]: Please try to not always instance a new mark instance if possible. I was able to do so in our createShortcutModal instance

**Review by JanAckermann — CHANGES_REQUESTED** [review]

**Review by JanAckermann — CHANGES_REQUESTED** [review]
Unfortunately you missed the trashbin adjustments :(

**Comment by JammingBen** [comment]
> Unfortunately you missed the trashbin adjustments :(

The acceptance criteria are missing it :P But yes, you're totally right

File: packages/web-app-files/src/views/shares/SharedWithMe.vue [hunk 2]
```diff
@@ -166,9 +182,28 @@ export default defineComponent({
         )
       }
 
+      if (unref(filterTerm).trim()) {
+        const usersSearchEngine = new Fuse(result, { ...defaultFuseOptions, keys: ['name'] })
+        result = usersSearchEngine.search(unref(filterTerm)).map((r) => r.item)
```
Comment (dschmidt) [inline packages/web-app-files/src/views/shares/SharedWithMe.vue:187]: The search order is apparently unstable like this, if that is desired you need to make it a filter on the original items, see #10046

**Review by dschmidt — APPROVED** [review]

**Review by JanAckermann — APPROVED** [review]

**Pull request merged**